ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -872,6 +896,31 @@ def test_frontend_function(
                 ):
                     assert ret.ivy_array.data is out.ivy_array.data
                 assert ret is out
+        elif test_flags.with_copy:
+            assert not isinstance(ret, tuple)
+
+            if test_flags.generate_frontend_arrays:
+                assert _is_frontend_array(ret)
+            else:
+                assert ivy_backend.is_array(ret)
```

Comment:
Arrays returned by frontend functions are always frontend arrays. This is being asserted [here](https://github.com/unifyai/ivy/blob/bae0fdfe24630f466ea5a474b8576d4ca717f502/ivy_tests/test_ivy/helpers/function_testing.py#L844). So this `if` statement can be replaced by a simple `assert _is_frontend_array(ret)`.

Reply:
Also, since you are asserting that `ret` is an array, I don't think you need to assert it is not a tuple as well.